src/backlight/portfolio/portfolio.py

```diff
@@ -90,14 +90,98 @@ def construct_portfolio(
     return Portfolio(positions)
 
 
-def calculate_pl(pt: Portfolio, mkt: List[MarketData]) -> pd.DataFrame:
+# def calculate_pl(pt: Portfolio, mkt: List[MarketData]) -> pd.DataFrame:
+#     """
+#     Apply the sum on the homogenized portfolio
+#     args:
+#         - portfolio : a defined portfolio
+#         - mkt : list of marketdata for each asset
+#         - base_ccy : asset of reference
+
+#     """
+#     pl = pt.value()
+#     return pl.sum(axis=1)
+
+
+def homogenize_pl(
+    pt: Portfolio, mkt: List[MarketData], base_ccy: str = "USD"
+) -> Portfolio:
+    """
+    Normalize PL to one asset reference and sum
+    args:
+       - portfolio : a defined portfolio
+       - mkt : list of marketdata for each asset
+       - base_ccy : asset of reference, all assets are converted this one
+    """
+    new_positions = []
+
+    for position in pt._positions:
+        ccy = position.symbol[-3:]
+        if ccy == base_ccy:
+            new_positions.append(
+                position.loc[mkt[0].index.intersection(pt._positions[0].index)].copy()
+            )
+        else:
+            new_positions.append(convert_positions(position, mkt, ccy, base_ccy))
+
+    return Portfolio(new_positions)
+
+
+def convert_positions(
```

Comment:
Make private function. Also, `convert_positions` is ambiguous.
How about naming it as `_convert_currency_unit`?

Reply:
If possible, I am very appreciate to add unittest.